lib/hpke/src/certificate.cpp

```diff
@@ -236,14 +284,14 @@ struct Certificate::ParsedCertificate
 
   Certificate::ExpirationStatus expiration_status() const
   {
-    auto* not_before = X509_get_notBefore(x509.get());
-    auto* not_after = X509_get_notAfter(x509.get());
+    auto* nb = X509_get_notBefore(x509.get());
+    auto* na = X509_get_notAfter(x509.get());
```

Comment:
I would replace this method with comparison based on the cached `std::chrono` times:

```
auto now = std::chrono::system_clock::now();
return (not_before < now) && (now < not_after);
```